lightning/src/routing/router.rs

```diff
@@ -2553,15 +2637,15 @@ where L::Target: Logger {
 	selected_route.sort_unstable_by_key(|path| {
 		let mut key = [CandidateHopId::Clear((42, true)) ; MAX_PATH_LENGTH_ESTIMATE as usize];
 		debug_assert!(path.hops.len() <= key.len());
-		for (scid, key) in path.hops.iter() .map(|h| h.0.candidate.id(true)).zip(key.iter_mut()) {
+		for (scid, key) in path.hops.iter() .map(|h| h.0.candidate.id(Some(true))).zip(key.iter_mut()) {
 			*key = scid;
 		}
 		key
 	});
 	for idx in 0..(selected_route.len() - 1) {
 		if idx + 1 >= selected_route.len() { break; }
-		if iter_equal(selected_route[idx    ].hops.iter().map(|h| (h.0.candidate.id(true), h.0.node_id)),
-		              selected_route[idx + 1].hops.iter().map(|h| (h.0.candidate.id(true), h.0.node_id))) {
+		if iter_equal(selected_route[idx    ].hops.iter().map(|h| (h.0.candidate.id(Some(true)), h.0.candidate.target().unwrap_or(maybe_dummy_payee_node_id))),
+		              selected_route[idx + 1].hops.iter().map(|h| (h.0.candidate.id(Some(true)), h.0.candidate.target().unwrap_or(maybe_dummy_payee_node_id)))) {
```

Comment:
I think we can just not unwrap here and have `iter_equal` compare against an `Option`.

@TheBlueMatt Do we even need the target here?